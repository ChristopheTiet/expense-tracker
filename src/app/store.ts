import { configureStore } from "@reduxjs/toolkit";
import expensesReducer from "../ducks/expenses-slice";
import filterReducer from "../ducks/filter-slice";

export const store = configureStore({
  reducer: {
    expenses: expensesReducer,
    filter: filterReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
