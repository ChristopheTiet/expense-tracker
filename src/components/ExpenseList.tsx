import { useAppSelector } from "../app/hooks";
import { selectExpenses } from "../app/selectors";
import ExpenseListItem from "./ExpenseListItem";

export default function ExpenseList() {
  const expenses = useAppSelector(selectExpenses);
  return (
    <div className="space-y-2 py-4">
      {expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        expenses.map((expenseItem) => {
          return <ExpenseListItem key={expenseItem.id} {...expenseItem} />;
        })
      )}
    </div>
  );
}
