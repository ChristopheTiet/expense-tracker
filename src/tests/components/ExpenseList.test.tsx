import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import ExpenseList from "../../components/ExpenseList";
import { store } from "../../app/store";
import { BrowserRouter } from "react-router-dom";

test("should render ExpenseList correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <ExpenseList />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
