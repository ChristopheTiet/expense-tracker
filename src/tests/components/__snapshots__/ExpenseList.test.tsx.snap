// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList correctly 1`] = `
<div
  className="space-y-2 py-4"
>
  <p>
    No expenses
  </p>
</div>
`;
